gene/database/dynamodb.py

```diff
@@ -361,22 +365,120 @@ def update_record(self, concept_id: str, field: str, new_value: Any,
+                except ClientError as e:
+                    raise DatabaseReadException(e)
+                records = response["Items"]
+                if not records:
+                    break
+                for record in records:
+                    batch.delete_item(Key={
+                        "label_and_type": record["label_and_type"],
+                        "concept_id": record["concept_id"]
+                    })
+
+    def delete_source(self, src_name: SourceName) -> None:
+        """Delete all data for a source. Use when updating source data.
+
+        :param src_name: name of source to delete
+        :raise DatabaseReadException: if DB client requires separate read calls and
+            encounters a failure in the process
+        :raise DatabaseWriteException: if deletion call fails
+        """
+        while True:
+            try:
+                response = self.genes.query(
+                    IndexName="src_index",
+                    KeyConditionExpression=Key("src_name").eq(
+                        src_name.value
+                    )
+                )
+            except ClientError as e:
+                raise DatabaseReadException(e)
+            records = response["Items"]
+            if not records:
+                break
+            with self.genes.batch_writer(
+                overwrite_by_pkeys=["label_and_type", "concept_id"]
+            ) as batch:
+                for record in records:
+                    try:
+                        batch.delete_item(Key={
+                            "label_and_type": record["label_and_type"],
+                            "concept_id": record["concept_id"]
+                        })
+                    except ClientError as e:
+                        raise DatabaseWriteException(e)
 
-    def flush_batch(self):
-        """Flush internal batch_writer."""
+        try:
+            self.metadata.delete_item(Key={
+                "src_name": src_name.value
+            })
+        except ClientError as e:
+            raise DatabaseWriteException(e)
+
+    def complete_write_transaction(self) -> None:
+        """Conclude transaction or batch writing if relevant."""
         self.batch.__exit__(*sys.exc_info())
         self.batch = self.genes.batch_writer()
+
+    def close_connection(self) -> None:
+        """Perform any manual connection closure procedures if necessary."""
+        self.batch.__exit__(*sys.exc_info())
+
+    def load_from_remote(self, url: Optional[str] = None) -> None:
+        """Load DB from remote dump. Not available for DynamoDB database backend.
+
+        :param url: remote location to retrieve gzipped dump file from
+        """
+        raise NotImplementedError
+
+    def export_db(self, export_location: Path) -> None:
+        """Dump DB to specified location. Not available for DynamoDB database backend.
+
+        :param export_location: path to save DB dump at
+        """
+        raise NotImplementedError
```

Comment:
If we eventually want to support this, we should make issues for each

Reply:
Yeah, I thought it might be tricky since the location of the shared-local DB file isn't directly available to the normalizer. Maybe we could add kwargs to this method and require the user specify that path themselves.